skills/src/main/java/cm/aptoide/skills/SkillsViewModel.kt

```diff
@@ -115,6 +121,34 @@ class SkillsViewModel(
         }
   }
 
+  fun validateUrl(uriString: String): UriValidationResult {
+    val uri: Uri = Uri.parse(uriString)
+    if (hasInvalidRequestStructure(uriString, uri)) {
+      return UriValidationResult.Invalid(RESULT_INVALID_URL)
+    }
+    if (usernameContainsInvalidCharacters(uri)) {
+      return UriValidationResult.Invalid(RESULT_INVALID_USERNAME)
+    }
+    return UriValidationResult.Valid(uri)
+  }
+
+  private fun usernameContainsInvalidCharacters(eskillsUri: Uri): Boolean {
+    val pattern = Pattern.compile("[^A-Za-z0-9_ ]+")
+    val username = eskillsUri.getQueryParameter("user_name")!!
+    val matcher: Matcher = pattern.matcher(username)
+    return matcher.find()
+
+  }
+
+  private fun hasInvalidRequestStructure(uriString: String, parsedUri: Uri): Boolean {
+    val parametersString = uriString.split("?")[1]
+    if (parametersString.contains("#")) {
+      return true
+    }
+    val parametersArray = parametersString.split("&")
+    return parsedUri.queryParameterNames.size != parametersArray.size
+  }
+
```

Comment:
The ViewModel doesn't need to know about this logic.
I would create a UseCase for the validateUrl method, so these 3 methods wont be in the ViewModel anymore